### PR TITLE
✨ Add Sequelize based migrations

### LINKED ISSUE #31: ✨ Provide migrations
#### Describe your issue

We need to support database migration for our sequelize-based repositories.
The `sequelize-cli` can accomplish this easily and efficiently.

The migration needs to support the following scenarios:
- Migrate with postgres
- Migrate with sqlite
- Migrate when all tables are located in the same database
- Migrate when the tables are spread across multiple databases

For the first version, it is sufficient to assume that all databases use the same technology, since BPMN Studio only uses sqlite anyway.

#### Issue checklist

Please check the boxes in this list after submitting your Issue:

- [x] I've checked if this issue already exists
- [x] I've included all the information that i think is relevant
- [x] I've added logs and/or screenshots (if applicable)
- [x] I've mentioned PRs and issues that relate to this one

### PR DESCRIPTION
**Pull request by ElRaptorus**
**Changes:**

- Add scripts and configs for sequelize-based migrations, specific for the meta repository.
- These migrations currently support migrating all the sqlite storages from the old specs as implemented by the Skeleton-Electron App to the new Runtime-Specs
- Supports all currently used storages: process_models, flow_node_instances and timers
- Each storage has its own folder and its own migration scripts within the `sequelize/migrations` folder

The migration is done by [Umzug](https://github.com/sequelize/umzug) and can be run by the backend itself. 

No further configs are required.

Currently, migration is executed, when the backend is first launched. To allow the BPMN Studio, or anybody else, to specify a custom path to the Sqlite storages, the startup is no longer executed automatically.
Instead, a function is returned, which accepts the sqlite storage path as a parameter.

To compensate for possible function loss, a `bin/index.js` was added, which starts up the backend without providing a custom sqlite path.

**Issues:**
Closes #31 

**PR:** #30

## How can others test the changes?

- Clear all sqlite databases you might have
- Start the latest Skeleton-Electron app in order to populate some database in the old design
- Fill those databases with data (Especially FlowNodeInstances and ProcessTokens)
- Run the migration script via `npm run migrate-postgres` or `npm run migrate-sqlite`
- See that all data was migrated successfully.

**NOTE ON SQLITE**:
Keep in mind that the Skeleton-Electron App stored its databases in a different location than the runtime!

## PR-Checklist

Please check the boxes in this list after submitting your PR:

- [x] You can merge this PR **right now** (if not, please prefix the title with "WIP: ")
- [x] I've tested **all** changes included in this PR.
- [x] I've also reviewed this PR myself before submitting (e.g. for scrambled letters, typos, etc.).
- [x] I've rebased the `develop` branch with my branch before finishing this PR.
- [x] I've **summarized all changes** in a list above.
- [x] I've mentioned all **PRs, which relate to this one**.
- [x] I've prefixed my Pull Request title is according to [gitmoji guide](https://gitmoji.carloscuesta.me/).

## Example

<details>
<summary>
:warning: Before merging please click here to expand and follow the guide.
</summary>
<br>

Please use `:twisted_rightwards_arrows:` at the beginning of your merge commit title.

Example 1:

<pre>
<code>
:twisted_rightwards_arrows: :bug: Fix Wrong Text Decoration at ...
</code>
</pre>

To get your commit message, just copy the first part of this pull request.

Example 2:
<pre>
<code>
**Changes:**

- Fixes Wrong Text Decoration at ...
- Fixes some typos
- ...

**Issues:**

Closes #NumberOfFixedIssue

PR: #NumberOfThisPR
</code>
</pre>
</details>

## Emojis

<details>
<summary>
Expand for a list of most used Emojis.
</summary>
<br>

Please prefix your commit messages with an Emoji.

Ref: https://gitmoji.carloscuesta.me/

| Description              | Glyphe               | Emoji  |
|--------------------------|----------------------|--------|
| Bugfix                   | `:bug:`              | 🐛     |
| Configuration releated   | `:wrench:`           | 🔧     |
| Cosmetic                 | `:lipstick:`         | 💄     |
| Dependencies Downgrade   | `:arrow_down:`       | ⬇️     |
| Dependencies Upgrade     | `:arrow_up:`         | ⬆️     |
| Formatting               | `:art:`              | 🎨     |
| Improving Performance    | `:zap:`              | ⚡️     |
| Initial commit           | `:tada:`             | 🎉     |
| Linter                   | `:rotating_light:`   | 🚨     |
| Miscellaneous            | `:package:`          | 📦     |
| New Feature              | `:sparkles:`         | ✨     |
| Refactoring Code         | `:recycle:`          | ♻️     |
| Releasing / Version tags | `:bookmark:`         | 🔖     |
| Removing Stuff           | `:fire:`             | 🔥     |
| Tests                    | `:white_check_mark:` | ✅     |
| Work In Progress (WIP)   | `:construction:`     | 🚧     |

</details>

### CODE REVIEW

File: sequelize/migrations/flow_node_instances/4.2.0-4.3.0-flow_node_instances.js [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+      name: 'FK_process_token_flow_node_instance',
+      references: {
+        table: 'FlowNodeInstances',
+        field: 'flowNodeInstanceId',
+      },
+      onDelete: 'cascade',
+    });
+    logger.info('Migration successful!');
+  },
+  down: async (queryInterface, Sequelize) => {
```
Comment (roblen45) [inline sequelize/migrations/flow_node_instances/4.2.0-4.3.0-flow_node_instances.js]: Muss in down nicht immer der "Rückweg" rein?

File: sequelize/migrations/process_models/4.2.0-4.3.0.js [hunk 2]
```diff
@@ -0,0 +1,17 @@
+'use strict';
```
Comment (roblen45) [inline sequelize/migrations/process_models/4.2.0-4.3.0.js]: Braucht man immer eine leere Datei?

File: sequelize/migrations/timers/4.2.0-4.3.0_timers.js [hunk 3]
```diff
@@ -0,0 +1,17 @@
+'use strict';
```
Comment (roblen45) [inline sequelize/migrations/timers/4.2.0-4.3.0_timers.js]: s.o.

File: sequelize/sqlite.json [hunk 4]
```diff
@@ -0,0 +1,13 @@
+{
+  "info": "This file is auto generated and required by the sequelize-cli. Please do not remove.",
+  "username": null,
+  "password": null,
+  "database": null,
+  "host": null,
+  "port": null,
+  "dialect": "sqlite",
+  "storage": "/Users/christianwerner/Library/Application Support/process_engine_runtime/databases/flow_node_instances.sqlite",
```
Comment (roblen45) [inline sequelize/sqlite.json]: Works on my machine?

Comment (ElRaptorus) [reply sequelize/sqlite.json]: Das ganze ist eine temporäre Config, die von der sequelize CLI benötigt wird und bei der Script ausführung automatisch generiert wird. D.h., wenn du die Migrations bei dir laufen lässt, steht hier was anderes.

Ich hab die Datei zur .gitignore hinzugefügt. Sorry für die Verwirrung.

File: sequelize/migrations/flow_node_instances/4.2.0-4.3.0-flow_node_instances.js (hunk 1 above)
Comment (ElRaptorus) [reply sequelize/migrations/flow_node_instances/4.2.0-4.3.0-flow_node_instances.js]: Brauchen wir sowas denn?

File: sequelize/migrations/process_models/4.2.0-4.3.0.js (hunk 2 above)
Comment (ElRaptorus) [reply sequelize/migrations/process_models/4.2.0-4.3.0.js]: Ja, sonst bricht die CLI mit einem Fehler ab.

File: sequelize/migrations/timers/4.2.0-4.3.0_timers.js (hunk 3 above)
Comment (ElRaptorus) [reply sequelize/migrations/timers/4.2.0-4.3.0_timers.js]: s.o.

File: sequelize/migrations/process_models/4.2.0-4.3.0.js (hunk 2 above)
Comment (ElRaptorus) [reply sequelize/migrations/process_models/4.2.0-4.3.0.js]: Tatsächlich braucht man die nicht. Hab ich entfernt.

**Comment by ElRaptorus** [comment]
https://github.com/process-engine/process_engine_runtime/issues/31

**Pull request merged**